modules/metadata/mod_remoteip.c

```diff
@@ -571,14 +597,99 @@ static int remoteip_modify_request(request_rec *r)
+            if (config->forbid_direct) {
+                /* Whoops, trusted internal proxy was not detected and direct requests are forbidden, deny the request
+                 */
+                return HTTP_FORBIDDEN;
+            }
+        } else {
+            /* Internal trusted proxy is connecting, perform request modification if any options for it are set
+             */
+            char *val;
+
+            /* Remote host option, applied directly to the request host header
+             */
+            if (config->host_header_name) {
+                const char *host;
+                if ( (host = apr_table_get(r->headers_in, config->host_header_name)) ) {
+                    /* Propagate host header value from the header set by proxy and update virtual host chosen (the primary purpose of it)
+                     */
+                    apr_array_header_t *hdrs = apr_array_make(r->pool, 0, sizeof(char*));
+                    while (*host && (val = ap_get_token(r->pool, &host, 1))) {
+                        *(char **)apr_array_push(hdrs) = apr_pstrdup(r->pool, val);
+                        if (*host != '\0')
+                            ++host;
+                    }
+                    apr_table_setn(r->headers_in, "Host", apr_pstrdup(r->pool, ((char **)hdrs->elts)[((hdrs->nelts)-1)]));
+                    r->hostname = apr_pstrdup(r->pool, ((char **)hdrs->elts)[((hdrs->nelts)-1)]);
+                    ap_update_vhost_from_headers(r);
+                }
+            }
+
+            /* Remote port option, due to the transient nature of subrequests we have to annotate the connection itself for ap_default_port() hook
+             */
+            if (config->port_header_name) {
+                const char *port;
+                if ( (port = apr_table_get(r->headers_in, config->port_header_name) )) {
+                    r->parsed_uri.port = r->server->port;
+                    apr_table_setn(r->connection->notes, "remoteip_port", apr_pstrdup(r->connection->pool, port));
+                } else {
+                    /* Subsequent requests inside a single connection need to have the annotation reset if they do not have the header set
+                     */
+                    apr_table_unset(r->connection->notes, "remoteip_port");
+                }
+            }
+
+            /* Remote HTTPS protocol option, due to transient nature of subrequests we have to annotate the connection itself for ap_conn_is_ssl() hook
+             */
+            if (config->proto_header_name) {
+                const char *proto;
+                if ( (proto = apr_table_get(r->headers_in, config->proto_header_name)) ) {
+                    /* Exact HTTPS protocol ID option not set means just the header presence is enough, otherwise check for a match
+                     */
+                    if (!config->proto_https_enable || !strcmp(proto, config->proto_https_enable)) {
```

Comment:
So if RemoteHTTPSEnableProto is not set we'll consider any RemoteProtoHeader value is https?

Reply:
This one is done so we may make virtual "HTTPS" mode dependent just on specific header being present or absent.

If we do default value of "https", then we need additional flag added like RemoteHTTPSEnableCheckProto (default to on, allowing setting it to off to enable "header presence" mode).

Reply:
Ack for ap_cstr_casecmp()

Reply:
Indeed, would simplify things for part of typical use cases. Probably can be done like `RemoteHTTPSEnableProto ""` then.

Reply:
If prefer`"*"` if you ask me :)
It's more explicit as a value and still quite unlikely as a sensible remote proto header valeur..

Reply:
Yes, "*" looks more clear to me as well on a second thought.